### PR TITLE
Ability to claim and withdraw accrued fees

### PR DESCRIPTION
**Pull request by jijordre**
Implementation of #92.

### CODE REVIEW

File: src/commands/claim.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+'use strict';
+
+module.exports = {
+    command: 'claim <nii|fees>',
+    describe: 'Display information about specified resource',
```
Comment (hubiierik) [inline src/commands/claim.js]: Is this the correct description?

Comment (jijordre) [reply src/commands/claim.js]: Indeed not. Will correct.

**Review by hubiierik — CHANGES_REQUESTED** [review]

File: src/commands/claim-commands/fees.js [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+            if (withdrawableAmount.gt(0)) {
+                spinner.start(`Withdrawing ${ethers.utils.formatUnits(withdrawableAmount, tokenInfo.decimals)} ${tokenInfo.symbol}`);
+                const withdrawableMonetaryAmount = await nahmii.MonetaryAmount.from(withdrawableAmount, tokenInfo.currency);
+                const withdrawTx = await claimant.withdrawFees(wallet, withdrawableMonetaryAmount, options);
+
+                await provider.getTransactionConfirmation(withdrawTx.hash, timeout);
+                spinner.succeed(`Withdrawal of ${ethers.utils.formatUnits(withdrawableAmount, tokenInfo.decimals)} ${tokenInfo.symbol} confirmed`);
+            }
+            else {
+                spinner.succeed('Withdrawal skipped');
```
[the remark this answers is not in the document]
Comment (jijordre) [reply src/commands/claim-commands/fees.js]: You may be right.

Now the full output is as follows if there is nothing to claim in the current session and also no amount previously claimed and not withdrawn:
```
> nahmii claim fees for NII --blocks=7171094
✔ Claimable amount of NII is 0.0
✔ Previously claimed (and not withdrawn) amount of NII is 0.0
✔ Claim skipped
✔ Withdrawal skipped
```

Possibly it should be updated to the following:
```
> nahmii claim fees for NII --blocks=7171094
✔ Claimable amount of NII is 0.0
✔ Nothing to claim
[✔ Previously claimed (and not withdrawn) amount of NII is 0.0]
✔ Withdrawable amount of NII is 0.0
✔ Nothing to withdraw
```
Here I've put the statement of previously claimed amount in brackets to suggest that it only is displayed if the amount is non-zero. In most cases the previously claimed amount is indeed 0, thus that statement is mostly just noise. In the actual scenario above it should _not_ appear in the flow of log statements.

File: src/commands/claim-commands/fees.js [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+
+            if (claimableAmount.gt(0)) {
+                spinner.start(`Claiming ${ethers.utils.formatUnits(claimableAmount, tokenInfo.decimals)} ${tokenInfo.symbol}`);
+                const claimAndStageTx = await range.claimFeesFn.call(claimant, wallet, currency, range.start, range.end, options);
+
+                await provider.getTransactionConfirmation(claimAndStageTx.hash, timeout);
+                spinner.succeed(`Claim of ${ethers.utils.formatUnits(claimableAmount, tokenInfo.decimals)} ${tokenInfo.symbol} confirmed`);
+            }
+            else {
+                spinner.succeed('Claim skipped');
```
[the remark this answers is not in the document]
Comment (jijordre) [reply src/commands/claim-commands/fees.js]: Answered by comment above.

File: src/commands/claim-commands/fees.js [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+            throw new Error(`Claiming of fees failed: ${err.message}`);
+        }
+        finally {
+            provider.stopUpdate();
+        }
+    }
+};
+
+function validateGasPrice(priceGWei) {
+    const price = parseInt(priceGWei);
```
[the remark this answers is not in the document]
Comment (jijordre) [reply src/commands/claim-commands/fees.js]: Should not be necessary really. The typical value is double digit only.

File: src/commands/claim-commands/fees.js [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+    if (price <= 0)
+        throw new Error('Gas price must be a number higher than 0.');
+    return price;
+}
+
+function validateGasLimit(gas) {
+    const gasLimit = parseInt(gas);
```
[the remark this answers is not in the document]
Comment (jijordre) [reply src/commands/claim-commands/fees.js]: As the current gas limit on mainnet is around 8m also here big number is not needed.

**Review by hubiierik — APPROVED** [review]

**Pull request merged**